src/D2L.Security.OAuth2/Provisioning/Default/AuthServiceClient.cs

```diff
@@ -168,16 +175,32 @@ IEnumerable<Scope> scopes
 		}
 
 		private sealed class GrantResponse {
-			[JsonProperty( PropertyName = "access_token", Required = Required.Always )]
+			[JsonPropertyName( "access_token" )]

```

Comment:
"required" things aren't supported in System.Text.Json.
I added `Validate()` methods and called them in the only places these types were constructed (outside of tests)
Note that these two classes are `private`.

Reply:
Hmm, so with `System.Text.Json` there's no way to differentiate between `"foo": null` and the `foo` property not existing at all?

Reply:
Not by default, at least.

IIRC it's more general: `"foo": 0` is equivalent to a missing `foo` if the target type is an `int` (i.e. its about `default`).